src/agent/agent.c

```diff
@@ -516,6 +516,10 @@ bool agent_set_port(Agent *agent, const char *port_s) {
         return true;
 }
 
+bool agent_set_manager_address(Agent *agent, const char *address) {
+        return copy_str(&agent->manager_address, address);
```

Comment:
Since we are copying the address here, we need to free it in `agent_unref` (with a preceding NULL check).

Reply:
You're right, and this is not a small comment. 
Just a question, AFAIK the new version of `free` is protected from NULL calls, isn't it? `copy_str` calls it without checking anything. So, do we need to check in `agent_unref` (we don't check any other call to `free`)?

Reply:
We are using gcc 11.4 as far as I know and valgrind will complain about an unnecessary `free` call - so I don't think it is protected. The other `free`s in `agent_unref` (and other parts) as well as the missing NULL checks are something we have to improve - in addition to setting them to NULL after freeing (see issue https://github.com/eclipse-bluechi/bluechi/issues/593).

Reply:
`free` must already be protected. Otherwise `copy_str` would have crashed at the start, in `free(*p)`:
```
static inline bool copy_str(char **p, const char *s) {
        char *dup = strdup(s);
        if (dup == NULL) {
                return false;
        }
        free(*p);
        *p = dup;
        return true;
}
```
Now, I'm not saying a cleaner code is not desirable. But, if we have a issue for it, not sure that fixing it just in this case is the right thing. Maybe a better fix is a macro that handles all three operations (check, free and NULL assignment)

Reply:
Ah, yes `free` on  `NULL` is perfectly fine. Sorry, I was mixing things up with calling `free` twice on the same pointer that hasn't been set to `NULL` after the first `free`.